From: Jiri Slaby <jslaby@suse.cz>
Subject: kABI: protect struct dm_exception_store_type
Patch-mainline: never, kabi

In 3.12.56, commit da7dbf91fceb737d8437055cd4ba4da57f971623 (dm
snapshot: fix hung bios when copy error occurs), upstream commit
385277bfb57faac44e92497104ba542cdd82d5fe changed prototype of
commit_exception in struct dm_exception_store_type. It indeed changed
the layout of the structure.

But given this is only dm-internal structure, protect the member the
usual way, by #ifdef __GENKSYMS__.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 drivers/md/dm-exception-store.h |    4 ++++
 1 file changed, 4 insertions(+)

--- a/drivers/md/dm-exception-store.h
+++ b/drivers/md/dm-exception-store.h
@@ -70,7 +70,11 @@ struct dm_exception_store_type {
 	 * Update the metadata with this exception.
 	 */
 	void (*commit_exception) (struct dm_exception_store *store,
+#ifdef __GENKSYMS__
+				  struct dm_exception *e,
+#else
 				  struct dm_exception *e, int valid,
+#endif
 				  void (*callback) (void *, int success),
 				  void *callback_context);
 
